pkg/client/monitoring/v1/types.go

```diff
@@ -215,6 +221,45 @@ type StorageSpec struct {
 	VolumeClaimTemplate v1.PersistentVolumeClaim `json:"volumeClaimTemplate,omitempty"`
 }
 
+// ThanosSpec defines parameters for a Prometheus server within a Thanos deployment.
+// +k8s:openapi-gen=true
+type ThanosSpec struct {
+	// Peers is a DNS name for Thanos to discover peers through.
+	Peers string `json:"peers"`
+	// Version describes the version of Thanos to use.
+	Version string `json:"version"`
+	// Thanos base image if other than default.
+	BaseImage *string `json:"baseImage,omitempty"`
+	// GCS configures use of GCS in Thanos.
+	GCS *ThanosGCSSpec `json:"gcs"`
+	// S3 configures use of S3 in Thanos.
+	S3 *ThanosS3Spec `json:"s3"`
+}
+
+// ThanosGCSSpec defines parameters for use of GCS with Thanos.
+// +k8s:openapi-gen=true
+type ThanosGCSSpec struct {
```

Comment:
why GCS / S3 distinction?

Reply:
Mostly because Thanos does, and I don't want to introduce an abstraction that could diverge if thanos diverges.

Reply:
if you want GCS it  requires  different configuration than if you want S3